### PR TITLE
Add `use_libtest_harness` attribute to rust_test

### PR DESCRIPTION
**Pull request by hlopko**
When set to False, rustc will not include libtest for the test,
therefore the test author is responsible for implementing `main`
function and running the tests.

### CODE REVIEW

File: rust/private/rust.bzl [hunk 1]
```diff
@@ -703,6 +703,13 @@ _rust_test_attrs = {
             ["Make variable"](https://docs.bazel.build/versions/master/be/make-variables.html) substitution.
         """),
     ),
+    "harness": attr.bool(
```
Comment (illicitonion) [inline rust/private/rust.bzl:706]: I'd suggest a slightly more verbose name, perhaps `use_libtest_harness`?

File: test/unit/test_harness/test_harness_test.bzl [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+def _test_harness_rustc_noharness_flags_test_impl(ctx):
+    env = analysistest.begin(ctx)
+    tut = analysistest.target_under_test(env)
+    action = tut.actions[0]
+    argv = action.argv
+    asserts.true(env, action.mnemonic == "Rustc", "expected the action to be the rustc invocation, got %s" % action.mnemonic)
+    asserts.true(env, "test/unit/test_harness/mytest_noharness.rs" in argv, "expected the action to build mytest.rs")
+    asserts.false(env, "--test" in argv, "expected rustc invocation not to contain --test, got %s" % len(argv))
+    asserts.true(env, "--cfg" in argv, "expected rustc invocation to contain --cfg, got %s" % len(argv))
+    asserts.true(env, "test" in argv, "expected rustc invocation to contain test, got %s" % len(argv))
```
Comment (illicitonion) [inline test/unit/test_harness/test_harness_test.bzl:27]: It would be nice to assert that these are adjacent, but not the end of the world not to

**Review by illicitonion — APPROVED** [review]

File: test/unit/test_harness/test_harness_test.bzl (hunk 2 above)
Comment (hlopko) [reply test/unit/test_harness/test_harness_test.bzl]: Well played :)

File: rust/private/rust.bzl (hunk 1 above)
Comment (hlopko) [reply rust/private/rust.bzl]: Done.

**Review by illicitonion — APPROVED** [review]
Thanks!

**Pull request merged**